src/net.cpp

```diff
@@ -2865,3 +2867,25 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(CNode* node, std::string& command, std::vector<unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node->addr.ToString();
+    boost::filesystem::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
+    std::ofstream f(path.c_str(), std::ios::binary | std::ios::out | std::ios::app);
```

Comment:
Could use `CAutoFile` instead (which provides proper serialization automatically)?

No need for `c_str()` here :)

Reply:
Thanks for suggesting this, `CAutoFile` is much nicer